galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.manager/src/main/java/dev/galasa/cicsts/internal/CicstsDefaultLogonProvider.java

```diff
@@ -52,10 +65,38 @@ public boolean logonToCicsRegion(ICicsTerminal cicsTerminal) throws CicstsManage
 
             waitForGmText(cicsTerminal);
 
-            cicsTerminal.clear();
-
             logger.debug("Logged onto " + cicsTerminal.getCicsRegion());
-        } catch (Zos3270Exception e) {
+
+            // If loginCredentials is provided, attempt to sign-in
+            // via CESL
+            if (!cicsTerminal.getLoginCredentials().isEmpty()) {
+                ICredentialsUsernamePassword creds = (ICredentialsUsernamePassword)this.cs.getCredentials(cicsTerminal.getLoginCredentials());
+                cts.registerText(creds.getPassword(), "Password for credentials");
```

Comment:
The 'reminder text' that is passed to the CTS should reference the tag that was used to obtain the creds. This is so the user can backtrack in the case of wrong credentials.  However thank you for adding this code!

Reply:
Added the credentials tag to the error message